client-src/elements/chromedash-feature-page.ts

```diff
@@ -119,6 +125,53 @@ export class ChromedashFeaturePage extends LitElement {
     return this.feature && Object.keys(this.feature).length !== 0;
   }
 
+  isFeatureUpcoming(latestStableVersion: number, stages: Array<StageDict>) {
+    if (!latestStableVersion || !stages) {
+      return;
+    }
+
+    const shippingMilestones = new Set<number | undefined>();
+    // Get milestones from all shipping stages, STAGE_TYPES_SHIPPING.
+    // TODO: Do we have to check stage.extensions?
+    for (const stage of stages) {
+      if (STAGE_TYPES_SHIPPING.has(stage.stage_type)) {
+        shippingMilestones.add(stage.desktop_first);
+        shippingMilestones.add(stage.android_first);
+        shippingMilestones.add(stage.ios_first);
+        shippingMilestones.add(stage.webview_first);
+      }
+    }
+    // Check if this feature is shipped within two milestones.
+    if (
+      shippingMilestones.has(latestStableVersion + 1) ||
+      shippingMilestones.has(latestStableVersion + 2)
+    ) {
+      this.isUpcoming = true;
+    }
+  }
+
+  /**
+   * A feature is outdated if it is scheduled to ship in the next 2 milestones,
+   * and its accurate_as_of date is at least 4 weeks ago.*/
+  isFeatureOutdated(): boolean {
+    if (!this.isUpcoming) {
+      return false;
```

Comment:
I'd like to show a warning if a feature hasn't been verified since it shipped, even if it claims to have shipped a year ago. That way readers know not to trust the page. I think the logic is something like:

```python
for each milestone in which the feature claims to be shipping or have shipped:
  if milestone <= latestStableVersion + 2:
    if accurateDate <= shipDate(milestone) && accurateDate + gracePeriod < this.currentDate:
      showWarning()
```

We should probably ask feature owners to re-verify the feature as soon as the ship-date passes, so they're not surprised by a warning 2 months later, but only warn other users after the grace period?

Reply:
ACK. Given the other comments in this PR, it looks like we will show two types of warnings:
1. if a feature will be shipped in upcoming milestones but outdated
2. if a feature has not been verified since it is shipped (this comment)

For the first warning, it will show tailored texts for feature authors and readers.

For the ease of reviews and testing, let me send up a follow-up PR for the second type.